custom_components/tuya_local/devices/smartmi_e1_purifier.yaml

```diff
@@ -0,0 +1,61 @@
+name: Smartmi Air Purifier E1
+products:
+  - id: pwrmggdhwkz1lnvh
+    name: Smartmi Air Purifier E1
+primary_entity:
+  entity: fan
+  translation_key: fan_with_presets
+  name: Air Purifier
+  icon: "mdi:air-purifier"
+  dps:
+    - id: 1
+      name: switch
+      type: boolean
+      mapping:
+        - dps_val: true
+          icon: "mdi:air-purifier"
+        - dps_val: false
+          icon: "mdi:air-purifier-off"
+    - id: 3
+      name: preset_mode
+      type: string
+      mapping:
+        - dps_val: sleep
+          value: Silent
```

Comment:
I strongly suggest adding `translation_key: fan_with_presets` to the entity definition above, and using preset names from https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/translations/en.json#L166 (the keys in the list under fan_wth_presets). ie. sleep, smart, normal, or if the original tuya dps_val are misleading sleep, normal, strong.